code.ts

```diff
@@ -1,21 +1,65 @@
+const clientStrageKey = 'team-library-components'
+
 async function main() {
-  if (figma.command == "saveTargetComponent") {
-    await saveTargetComponent()
+  if (figma.command == "saveComponent") {
+    await figma.clientStorage.setAsync(clientStrageKey, await saveComponent([figma.currentPage])) 
   }
   else if (figma.command == "replaceNodes") {
-    await replaceNodes()
+    const teamLibraryComponents = await figma.clientStorage.getAsync(clientStrageKey)    
+    await findTargetNodes(figma.currentPage.selection, teamLibraryComponents)
   }
   figma.closePlugin()
 }
 
-// ComponentFileのComponent IDを取得して保存
-async function saveTargetComponent() {
-  return 1
+async function saveComponent(nodes) {
+  const teamLibraryMasterComponents = {}
+  findComponent(teamLibraryMasterComponents, nodes)
+
+  return teamLibraryMasterComponents
+}
+
+async function findComponent(teamLibraryMasterComponents, nodes) {
+  for (const node of nodes) {
+    if (node.type === "COMPONENT") {
+      teamLibraryMasterComponents[node.name] = node.key
+    } 
+    if (node.children != null) {
+      findComponent(teamLibraryMasterComponents, node.children)
+    }
+  } 
 }
 
 // 保存したComponent IDを使ってリプレイス
-async function replaceNodes() {
-  return 1
+async function findTargetNodes(nodes, teamLibraryComponents) {
+  for (const node of nodes) {
+    if (node.type === "INSTANCE" || node.type === "FRAME" || node.type === "GROUP") {
+      const key = teamLibraryComponents[node.name]
+      if (key != undefined) { 
+        if (node.type === "INSTANCE" && node.mainComponent.key === teamLibraryComponents[node.name]) {
+          continue
+        }
+        await replaceNodes(node, key)
+      } else {
+        findTargetNodes(node.children, teamLibraryComponents)
+      }
+    } 
+  }
+}
+
+async function replaceNodes(node, key) {
+  try {  
+    const getTeamLibraryComponent = await figma.importComponentByKeyAsync(key)
```

Comment:
```suggestion
    const teamLibraryComponent = await figma.importComponentByKeyAsync(key)
```

getTeamLibraryComponent だとメソッド名みたいに見えてしまうかな〜